File: C++/binary_search-3a.cpp
#include <iostream>

int binary_search(int arr[], int size, int n) {
    int low = 0;
    int high = size - 1;
    int mid;

    while (high >= low) {
        mid = (high + low) / 2;

        if (arr[mid] == n) {
            return mid; 
        }

        if (arr[mid] < n)
            low = mid + 1;
        else
            high = mid - 1;
    }

    return -1; 
}

void output_array(int arr[], int size) {
    for (int i = 0; i < size; i++) 
    {
        std::cout << arr[i] << " ";
    }
    std::cout << std::endl;
}

int main() {
    int size;
    std::cout << "Enter size of array: ";
    std::cin >> size;

    int a[size];
    std::cout << "Enter all elements of sorted array: ";
    for (int i = 0; i < size; i++) {
        std::cin >> a[i];
    }

    std::cout << "The elements entered in sorted array are: ";
    output_array(a, size);

    int n;
    std::cout << "Enter n: ";
    std::cin >> n;

    int result = binary_search(a, size, n);

    if (result != -1) 
        std::cout << "Element " << n << " found at index " << result << std::endl;
    else 
        std::cout << "Element " << n << " not found in the array" << std::endl;

    return 0;
}

File: Javascript/binary_search-3a.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function binarySearch(arr, n) {
    let low = 0;
    let high = arr.length - 1;

    while (high >= low) {
        let mid = Math.floor((high + low) / 2);

        if (arr[mid] === n) {
            return mid;
        }

        if (arr[mid] < n) {
            low = mid + 1;
        } else {
            high = mid - 1;
        }
    }

    return -1;
}

function outputArray(arr) {
    console.log("The elements entered in sorted array are: " + arr.join(' '));
}

function main() {
    rl.question("Enter size of array: ", (size) => {
        size = parseInt(size);
        let a = [];

        console.log("Enter all elements of sorted array:");
        rl.on('line', (input) => {
            a.push(parseInt(input));

            if (a.length === size) {
                rl.close();
            }
        });

        rl.on('close', () => {
            outputArray(a);

            rl.question("Enter n: ", (n) => {
                n = parseInt(n);
                let result = binarySearch(a, n);

                if (result !== -1) {
                    console.log(`Element ${n} found at index ${result}`);
                } else {
                    console.log(`Element ${n} not found in the array`);
                }

                process.exit(0);
            });
        });
    });
}

main();
